torch/csrc/jit/passes/onnx.cpp

```diff
@@ -39,6 +40,40 @@ void RemovePrintOps(std::shared_ptr<Graph>& graph) {
   removePrintOps(graph->block());
 }
 
+Value* insertIValueWithoutTupleConstants(Graph* graph, const IValue& ival) {
```

Comment:
I feel like this sets the trend that each new constant type will need a new onnx pass to remove it from the graph.  That seems like much more work than simply putting a rule into onnx so that it knows how to deal with the tuple constants directly.  Depending on the actual issue it might be a good idea to teach tuple lowering about tuple constants, so that they get lowered appropriately.

Reply:
`# onnx does not support tuples, so try to remove them` - I don't think onnx handles tuples at all. i think it fails if there are any tuples you can't lower in the graph. I added this pass to fix a test that was failing.

Reply:
Maybe the test is failing because the LowerAllTuples pass doesn't correctly handle tuple constants. If so, it would be more concise to fix that pass, with the added benefit that we would be able to lower tuple constants there as well.

Reply:
Constant Propagation will already handle lowering of tuple constants

Reply:
When I implemented `str[]` and `float[]` constants, I communicated with onnx folks and they're adding support for exporting it. I don't think anytime someone adds constants to the JIT the burden is necessarily on them to also fix onnx representation, in this case it was an easy fix and less hassle so I implemented it.